lib/srv/desktop/windows_server.go

```diff
@@ -828,7 +840,7 @@ func (s *WindowsService) generateCredentials(ctx context.Context, username, doma
 	// CRLs in it. Each service can also handle RDP connections for a different
 	// domain, with the assumption that some other windows_desktop_service
 	// published a CRL there.
-	crlDN := "CN=Teleport,CN=CDP,CN=Public Key Services,CN=Services,CN=Configuration," + s.cfg.LDAPConfig.domainDN()
```

Comment:
FYI: the actual bug was here - we weren't adding the cluster name to the DN used in the certificate.

The rest of the changes are just cleanup and tests to help make sure this doesn't happen again.